crates/pallet-domains/src/staking.rs

```diff
@@ -92,11 +96,15 @@ pub(crate) struct PendingDeposit<Balance: Copy> {
 }
 
 /// A nominator's withdrawal from a given operator pool.
-#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
-pub(crate) struct Withdrawal<Share, DomainBlockNumber> {
-    pub(crate) allowed_since_domain_epoch: DomainEpoch,
-    pub(crate) unlock_at_confirmed_domain_block_number: DomainBlockNumber,
-    pub(crate) shares: Share,
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq, Default)]
+pub(crate) struct Withdrawal<Balance, Share, DomainBlockNumber> {
+    /// Total withdrawal amount requested by the nominator that are in unlocking state excluding withdrawal in shares.
+    pub(crate) total_withdrawal_amount: Balance,
```

Comment:
Why do we need to track the total? Is it only used to check underflow/overflow?

Reply:
The total is tracked so that we don't loop through each withdrawal amount in each epoch for each nominator while unlocking whole operator or slashing the operator. We just use the total